bellbot_gui/src/gui2.py

```diff
@@ -0,0 +1,282 @@
+          res = proxy(NewTargetRequest(dest))
+          return res
+          #return NewTargetResponse()
+        except rospy.ServiceException, e:
+          print "Service call failed: %s"%e
+
+
+    def sort_dests_per_categ(self, dests=None):
+        if dests is None:
+            dests = self.dests
+        dests_per_categ = {}
+        for c in self.categories:
+            dests_per_categ[c] = {}
+        for k, v in dests.items():
+            dests_per_categ[v.kind][v.id] = v
+            # print k, "//", v.id, "//", v.name, "//", v.kind
+        # with open("dests_per_categ.p", "wb") as f:
+        #     pickle.dump(dests_per_categ, f)
+        return dests_per_categ
+
+    def get_metadata(self):
+        dests = {}
+        rospy.wait_for_service('/query_node_metadata')
+        try:
+            proxy = rospy.ServiceProxy('/query_node_metadata', NodeMetadata)
+            for c in self.categories:
+                print "Getting", c
+                res = proxy(NodeMetadataRequest("aaf_bellbot", "aaf_bellbot", c)) # 'office' | 'Meeting Rooms'
+                for i in range(0, len(res.name)):
+                    foo = res.name[i]
+                    # foo = res.name[i].replace("-", "_")
+                    dests[foo] = Destination_Data(name=foo, description=res.description[i], kind=res.node_type[i], goto=res.goto_node[i], available=True, at_node=res.at_node[i])
+
+        except rospy.ServiceException, e:
+            print "Service call failed: %s"%e
+        return dests
+
+    def create_select_dest_page(self):
+        config_parser = ConfigParser.SafeConfigParser()
+        if len(config_parser.read(self.ini)) == 0:
+            raise IOError("ini file not found")
+        try:
+            gui_select_dest_header = os.path.join(self.http_root, config_parser.get("gui", "gui_select_dest_header"))
+            gui_select_dest_footer = os.path.join(self.http_root, config_parser.get("gui", "gui_select_dest_footer"))
+            select_dest_page_filename = config_parser.get("gui", "gui_select_dest_page")
+            gui_select_dest_page = os.path.join(self.http_root, select_dest_page_filename)
+        except ConfigParser.NoOptionError:
+            raise
+        with open(gui_select_dest_header, "r") as f:
+            data_gui_select_dest_header = f.read()
+        with open(gui_select_dest_footer, "r") as f:
+            data_gui_select_dest_footer = f.read()
+        options_str = ""
+        for categ, rooms in self.dests_per_categ.items():
+            options_str += '<optgroup label="' + categ + '">\n'
+            rooms_ids = sorted(rooms.keys())
+            for room_id in rooms_ids:
+                options_str += self.ret_options_str(rooms[room_id])
+            options_str += "</optgroup>\n"
+        with open(gui_select_dest_page, "w") as f:
+            f.write(data_gui_select_dest_header)
+            f.write(options_str)
+            f.write(data_gui_select_dest_footer)
```

Comment:
This will not work due to missing write permission in `/opt/ros/indigo/` where the http_root will be installed I guess.